pages/_app.js

```diff
@@ -2,13 +2,20 @@ import GlobalStyle from "@/styles";
 import Head from "next/head";
 import useLocalStorageState from "use-local-storage-state";
 import userDevices from "../db.json";
+import { v4 as uuidv4 } from "uuid";
 
 export default function App({ Component, pageProps }) {
   const [devices, setDevices] = useLocalStorageState("devices", {
     defaultValue: [...userDevices],
   });
-  function createDevice(device) {
-    setDevices([...devices, device]);
+  function createDevice(newDevice) {
+    setDevices((oldDevices) => [
+      {
+        ...newDevice,
+        id: uuidv4(),
+      },
+      ...oldDevices,
+    ]);
   }
 
```

Comment:
Why do you handle your createDevice-function in the _app-page?

Reply:
Because later on, I need the devices on multiple pages.